src/scripts.js

```diff
@@ -281,24 +281,54 @@ function displaySelectedRecipe(e) {
       return currentRecipe.id === recipeID;
     })
 
-    show([selectedRecipeView, favoritePageButton, cookbookPageButton]);
+    show([selectedRecipeView, favoritePageButton, cookbookPageButton, pantryPageButton]);
     hide([mainView, searchBar, searchButton, pageTitle, pantryView, highlightKey]);
     showCookbookStatus(selectedRecipe);
     showFavoritesStatus(selectedRecipe);
     updateRecipeText(e, selectedRecipe, ingredientsData);
+
+    displayIngredientsNeeded(selectedRecipe);
+  }
+}
+
+function displayIngredientsNeeded(recipe) {
+  const neededIngredients = currentUser.pantry.getMissingIngredients(recipe);
+  const neededIngredientsSection = document.querySelector('.ingredients-needed-js');
+
+  if (neededIngredients.length === 0) {
+    neededIngredientsSection.innerText = 'you have all of the ingredients needed to cook this recipe!';
+  } else {
+    const elements = neededIngredients.reduce((acc, ingredient) => {
+      const matchedId = ingredientsData.find(item => {
+        return item.id === ingredient.ingredient
+      })
+      const ingWithUnits = selectedRecipe.ingredients.find(item => {
+        return item.id === ingredient.ingredient
+      })
+
+      acc += `
+        ${matchedId.name}: ${ingredient.amount} ${ingWithUnits.quantity.unit}<br>
+      `
+      return acc;
+    }, '<h3>for this recipe you are missing: </h3>')
+    neededIngredientsSection.innerHTML = elements;
   }
 }
 
+
 function displayPantryView() {
-  whatsCookin.classList.remove('home-page');
-  pageTitle.innerText = "my pantry";
-  show([pantryView]);
-  hide([mainView, searchBar, searchButton]);
-  populatePantry();
+  if (pantryView.classList.contains('hidden')) {
+    whatsCookin.classList.remove('home-page');
+    pageTitle.innerText = "my pantry";
+    show([pantryView, cookbookPageButton, favoritePageButton]);
+    hide([mainView, searchBar, searchButton, pantryPageButton, selectedRecipeView]);
+    populatePantry();
+  }
 }
 
 function populatePantry() {
   const tableBody = document.querySelector('tbody')
+  tableBody.innerHTML = '';
```

Comment:
Needed this because it was re-adding all the ingredients each time the 'my pantry' button was clicked.